### PR TITLE
Move programmatic navigation to "navigate" mirroring @reach/router + some fixes for prefetching

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6d566202ed837f2c5bb545

**Comment by KyleAMathews** [comment]
Deploy preview for *using-styled-components* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/using-styled-components/deploys/5b6d566302ed837f2c5bb551

**Comment by KyleAMathews** [comment]
Deploy preview for *using-jss* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/using-jss/deploys/5b6d566602ed837f2c5bb56c

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/using-glamor/deploys/5b6d566302ed837f2c5bb54e

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/using-contentful/deploys/5b6d566302ed837f2c5bb554

**Comment by KyleAMathews** [comment]
Deploy preview for *image-processing* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/image-processing/deploys/5b6d566202ed837f2c5bb54b

File: www/src/components/search-form.js [hunk 1]
```diff
@@ -232,47 +232,42 @@ css.insert(`
...
+    navigate(`${a.pathname}${a.hash}`)
   }
-
   componentDidMount() {
     if (
       typeof window === `undefined` || // eslint-disable-line no-undef
-      typeof window.docsearch === `undefined` // eslint-disable-line no-undef
+      typeof window.docsearch === `undefined`
     ) {
+      // eslint-disable-line no-undef
```
Comment (m-allanson) [inline www/src/components/search-form.js]: this should be bumped back up to line 260?

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 0a4b1016fad97e075504c9099ff5551aab901e5f

https://app.netlify.com/sites/gatsbyjs/deploys/5b6d566002ed837f2c5bb53a

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 42f40a15f892c6230aba885bf84d0ebd4fc1f211

https://app.netlify.com/sites/gatsbyjs/deploys/5b6d59e5fdd72a75f4b28932

**Pull request merged**